intern.go

```diff
@@ -117,8 +117,13 @@ func get(k key) *Value {
 	if valSafe != nil {
 		valSafe[k] = v
 	} else {
-		valMap[k] = uintptr(unsafe.Pointer(v))
+		// A note from bcmills: "... the uintptr conversion should happen after
+		// the call to runtime.SetFinalizer, not before. Even with the current
+		// (as of Go 1.15) memory model, the Value could be allocated on the
+		// stack before the finalizer is set, then moved to the heap when the
```

Comment:
If v is allocated on the stack, runtime.SetFinalizer might fail (as opposed to moving v to the stack). We could fashion some other means to force v to escape first. But I'm not entirely convinced it makes sense to complicate the code now to protect against complex hypothetical compiler optimizations.

Reply:
TIL. Would you prefer I leave the code as-is but add a quick comment noting the potential footgun described in #13? Or do you think this should just be closed out?

Reply:
I'm inclined to add this note from Bryan to the long "implementation details and dangers" comment near the end of the file. But this change doesn't do any harm, either. Let's wait for @bradfitz or @danderson to weigh in.

Thanks for following up on this!

Reply:
If you want to submit this, at least massively reduce this comment to something like:

```go
// call SetFinalizer before uintptr conversion (theoretical concern;
// see https://github.com/go4org/intern/issues/13)
```